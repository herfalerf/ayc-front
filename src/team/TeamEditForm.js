import React, { useState, useEffect } from "react";
import { useParams, useHistory } from "react-router-dom";
import { Formik, Form, Field } from "formik";
import { TextField } from "formik-material-ui";
import { Button, ButtonGroup, Grid, Typography } from "@material-ui/core";
import { Save, DeleteForever } from "@material-ui/icons/";
import { makeStyles } from "@material-ui/core/styles";
import ConfirmDialog from "../common/ConfirmDialog";
import * as yup from "yup";
import AycApi from "../api/api";

// Team member edit form
//
// Displays team member edit form and handles changes to local form state.
// Submitting the form calls the API to save
//
// Routed as /admin/team/:id

const useStyles = makeStyles((theme) => ({
  upper: {
    backgroundImage: `url('${process.env.PUBLIC_URL}/images/admin-home/admin-home01.png')`,
    backgroundSize: "cover",
    backgroundPosition: "center bottom",
    minHeight: "25vh",
  },
  form: {
    padding: theme.spacing(4),
    backgroundColor: "rgb(244,245,248)",
  },
  message: {
    color: "#00c7d7",
  },
}));

const TeamEditForm = ({ editMember, deleteMember }) => {
  const { id } = useParams();
  console.debug("TeamEditForm", "id=", id);
  const [member, setMember] = useState({
    name: "",
    title: "",
    bio: "",
    img: "",
  });
  const history = useHistory();
  const schema = yup.object().shape({
    name: yup.string().required("Name required"),
  });
  const classes = useStyles();

  useEffect(
    function getTeamMember() {
      async function getMember() {
        setMember(await AycApi.getMember(id));
      }

      getMember();
    },
    [id]
  );

  return (
    <div>
      <div className={classes.upper}></div>
      <Grid container justifyContent="center" className={classes.form}>
        <Grid item>
          <Typography variant="h5" className={classes.message}>
            Edit Team Member
          </Typography>
          <Formik
            validationSchema={schema}
            initialValues={{
              name: member.name,
              title: member.title,
              bio: member.bio,
              img: member.img,
            }}
            enableReinitialize={true}
            onSubmit={async (values, { setStatus, setSubmitting }) => {
              setStatus(undefined);
              console.debug("id=", id, "values=", values);
              let result = await editMember(id, values);
              console.debug(result);

              if (result.success) {
                setSubmitting(false);
                history.push("/admin/team");
              } else {
                setStatus({ error: result.errors });
              }
            }}
          >
            {({ isSubmitting, status }) => (
              <Form>
                <Field
                  label="name*"
                  type="name"
                  name="name"
                  component={TextField}
                />

                <br />
                <Field
                  label="title"
                  type="title"
                  name="title"
                  component={TextField}
                  InputProps={{ multiline: true }}
                />

                <br />
                <Field
                  label="bio"
                  type="bio"
                  name="bio"
                  component={TextField}
                  InputProps={{ multiline: true }}
                />

                <br />
                <Field
                  label="img"
                  type="img"
                  name="img"
                  component={TextField}
                  InputProps={{ multiline: true }}
                />

                {status && status.error ? (
                  <div>API Error: {status.error}</div>
                ) : null}
                <br />
                <ButtonGroup>
                  <Button
                    startIcon={<Save />}
                    variant="contained"
                    color="primary"
                    disabled={isSubmitting}
                    type="submit"
                  >
                    Submit
                  </Button>
                  <ConfirmDialog
                    startIcon={<DeleteForever />}
                    name="Delete"
                    title="Are you sure you want to delete?"
                    color="secondary"
                    onConfirm={async () => {
                      let result = await deleteMember(id);

                      if (result.success) {
                        history.push("/admin/team");
                      } else {
                        console.debug("API error", result.errors);
                      }
                    }}
                  ></ConfirmDialog>
                </ButtonGroup>
              </Form>
            )}
          </Formik>
        </Grid>
      </Grid>
    </div>
  );
};

export default TeamEditForm;
