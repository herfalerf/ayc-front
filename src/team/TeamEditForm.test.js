import React from "react";
import { render } from "@testing-library/react";
import TeamEditForm from "./TeamEditForm";
import { MemoryRouter } from "react-router";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <TeamEditForm />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <TeamEditForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
