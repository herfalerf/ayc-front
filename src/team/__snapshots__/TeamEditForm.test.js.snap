// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="makeStyles-upper-4"
    />
    <div
      class="MuiGrid-root makeStyles-form-5 MuiGrid-container MuiGrid-justify-content-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h5
          class="MuiTypography-root makeStyles-message-6 MuiTypography-h5"
        >
          Edit Team Member
        </h5>
        <form
          action="#"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              name*
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="name"
                type="name"
                value=""
              />
            </div>
          </div>
          <br />
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              title
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                name="title"
                rows="1"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
                tabindex="-1"
              />
            </div>
          </div>
          <br />
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              bio
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                name="bio"
                rows="1"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
                tabindex="-1"
              />
            </div>
          </div>
          <br />
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              img
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                name="img"
                rows="1"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
                tabindex="-1"
              />
            </div>
          </div>
          <br />
          <div
            class="MuiButtonGroup-root"
            role="group"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined MuiButton-containedPrimary"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                <span
                  class="MuiButton-startIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
                    />
                  </svg>
                </span>
                Submit
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeMedium"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
                      />
                    </svg>
                  </span>
                  Delete
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
